### ISSUE TITLE
console.log triggering in ng-material-multilevel-menu.js line:514

### ISSUE THREAD

**Issue by knpaing** [ShankyTiwari/ng-material-multilevel-menu#80]
Hi,

I just realised that console.log is left there in the mentioned file and had to comment out from printing the menu icons. Please see the attachment. And thank you very much for updating the menu. :)

<img width="1677" alt="Screenshot 2019-10-28 at 9 55 12 PM" src="https://user-images.githubusercontent.com/12625702/67691954-cfb70200-f9cd-11e9-9da5-fdacb948c55b.png">

**Comment by johnpankowicz** [comment]
I'm am somewhat glad the log messages were there. Otherwise I would not have seen that the code is calling this routine thousands of times to handle the five icons that I have in my menus! What is it doing and can't this be optimized?

**Comment by cityinspain** [comment]
@johnpankowicz It certainly shouldn't be running hundreds of thousands of times. Would you mind sharing your menu configuration? I'm having a difficult time reproducing this issue.

**Comment by johnpankowicz** [comment]
This is my first attempt to use ng-material-multilevel-menu, so I could be doing something very silly. I copied code from one of your samples and modified it's menu layout and left the same config. 

```
config = {
  paddingAtStart: true,
  interfaceWithRoute: true,
  classname: 'my-custom-class',
  listBackgroundColor: `rgb(208, 241, 239)`,
  fontColor: `rgb(8, 54, 71)`,
  backgroundColor: `rgb(208, 241, 239)`,
  selectedListFontColor: `blue`,
  highlightOnSelect: true,
  collapseOnSelect: true,
  rtlLayout: false
};

  appitems = [
    {
        label: 'GOVERNMENT',
        disabled: true
        // faIcon: 'fas fa-city'
        // imageIcon: '/assets/batman.jpg'
        // link: 'https://www.npmjs.com/package/ng-material-multilevel-menu',
        // externalRedirect: true
    },
    {
        label: 'City of Austin',
        faIcon: 'fas fa-city',
        items: [
            {
                label: 'City Council',
                link: '/item-1-1',
                faIcon: 'fab fa-accusoft'
            },
            {
                label: 'Board of Education',
                link: '/item-1-1',
                faIcon: 'fab fa-accessible-icon',
                disabled: false,
                onSelected: function() {
                  console.log('Item 1.2.2.1');
              }
            },
            {
              label: 'Planning Board',
              link: '/item-1-1',
              faIcon: 'fab fa-accessible-icon',
              disabled: false,
              onSelected: function() {
                console.log('Item 1.2.2.1');
            }
          }
      ]
    },

    {
      label: 'Travis County',
      faIcon: 'far fa-image',
      items: [
          {
              label: 'Commissioners',
              link: '/item-1-1',
              faIcon: 'fab fa-accusoft'
          },
          {
              label: 'Transportation',
              link: '/item-1-1',
              faIcon: 'fab fa-accessible-icon',
              disabled: false,
              onSelected: function() {
                console.log('Item 1.2.2.1');
            }
          }
    ]
  },

  {
    label: 'State of Texas',
    faIcon: 'fas fa-star',
    items: [
        {
            label: 'Senate',
            link: '/item-1-1',
            faIcon: 'fab fa-accusoft'
        },
        {
            label: 'House of Representatives',
            link: '/item-1-1',
            faIcon: 'fab fa-accessible-icon',
            disabled: false,
            onSelected: function() {
              console.log('Item 1.2.2.1');
          }
        }
  ]
},

{
  label: 'US Congress',
  faIcon: 'fas fa-flag',
  items: [
      {
          label: 'Senate',
          link: '/item-1-1',
          faIcon: 'fab fa-accusoft'
      },
      {
          label: 'House of Representatives',
          link: '/item-1-1',
          faIcon: 'fab fa-accessible-icon',
          disabled: false,
          onSelected: function() {
            console.log('Item 1.2.2.1');
        }
      }
]
},

{
  label: 'NON-GOVERNMENT',
  disabled: true
},

{
  label: 'Glendale HOA',
  faIcon: 'far fa-building',
  items: [
      {
          label: 'Governing Board',
          link: '/item-1-1',
          faIcon: 'fab fa-accusoft'
      },
      {
          label: 'Covenance Committee',
          link: '/item-1-1',
          faIcon: 'fab fa-accessible-icon',
          disabled: false,
          onSelected: function() {
            console.log('Item 1.2.2.1');
        }
      }
]
},
{
  label: 'Paws Rescue',
  faIcon: 'fas fa-paw',
  items: [
      {
          label: 'Trustees',
          link: '/item-1-1',
          faIcon: 'fab fa-accusoft'
      },
      {
          label: 'Special Meetings',
          link: '/item-1-1',
          faIcon: 'fab fa-accessible-icon',
          disabled: false,
          onSelected: function() {
            console.log('Item 1.2.2.1');
        }
      }
    ]
  },

  {
    label: 'DOCUMENTATION',
    disabled: true
  },

  ]


```

**Comment by johnpankowicz** [comment]
Here's my html:

```
<ng-material-multilevel-menu
    [configuration]='config'
    [items]='appitems'
    (selectedItem)="selectedItem($event)"
    (selectedLabel)="selectedLabel($event)">
</ng-material-multilevel-menu>

```

**Comment by cityinspain** [comment]
I implemented your configuration in the demo app in this repository. While there definitely are many messages being logged, I'm not noticing nearly the same amount that you suggested.

The function in question runs for each item in the menu and performs a check of whether or not its selected, and whether or not it has a separate active icon. Therefore, it will run once for each instance of a menu item with a Font Awesome icon. There really isn't much room for optimization.

If you provide the rest of your code in your own repo, I'd be happy to take a look, to see why this is occurring in your implementation.

(By the way, I'm not the author of this package or any of the samples, just a contributor)

**Comment by johnpankowicz** [comment]
Thanks for trying to help. I was about to create a stackblitz with the code. I first ran the component alone in a bare angular project. When I did this the number of log messages was done 10,000 fold. Now there were only 4 separate log messages for each fontawesome icon. In my main project, I was including the sidebar menu component in a CSS grid. I'll try removing some code at a time to try to figure out exactly when the number of log messages explodes.

**Comment by johnpankowicz** [comment]
I found the issue. ng-material-multilevel-menu has a conflict with amcharts. As soon as I removed the component that uses amcharts, the endless logging stopped. The particular chart that was being displayed was a "guage" that was being updated with a timeout. This amchart code was also just demo code off of their site as is the  ng-material-multilevel-menu sample code that I am using. 

amcharts is not using ng-material-multilevel-menu, angular material or even fontawesome icons, so I can't see how there should be a conflict. Also, the timeout values appear to be in seconds, so even if there were, it doesn't explain the 100s of thousands of log message that seem to be occurring.

I created a Github repo with the code that reproduces the problem:
[johnpankowicz/ngmenuchart](https://github.com/johnpankowicz/ngmenuchart)

I should probably create a new issue here for this problem
